pkg/app/piped/driftdetector/ecs/detector.go

```diff
@@ -480,3 +497,13 @@ func ignoreAutoScalingDiff(r *provider.DiffResult) bool {
 		r.New.ServiceDefinition.DesiredCount == 0 && // When desiredCount is 0 or not defined in the head manifest, autoscaling may be enabled.
 		r.Old.ServiceDefinition.DesiredCount != r.New.ServiceDefinition.DesiredCount
 }
+
+func sortKeyPairs(kps []types.KeyValuePair) []types.KeyValuePair {
+	sorted := make([]types.KeyValuePair, len(kps))
+	copy(sorted, kps)
```

Comment:
How about cloning here and removing `slices.Clone` when calling? That will reduce copying slices.

```suggestion
	sorted := slices.Clone(kps)
```

We can notice this function won't have side effects from a caller side because it returns slices.

Reply:
fixed in https://github.com/pipe-cd/pipecd/pull/5344/commits/930b46fc7f1fdf5523793cb7d503330cbdd6e7bb